### ISSUE TITLE
Platform-specific config

### ISSUE THREAD

**Issue by byronigoe** [ionic-team/capacitor#5295]
### Capacitor Version

```
Latest Dependencies:

  @capacitor/cli: 3.3.2
  @capacitor/core: 3.3.2
  @capacitor/android: 3.3.2
  @capacitor/ios: 3.3.2

Installed Dependencies:

  @capacitor/android: not installed
  @capacitor/ios: not installed
  @capacitor/cli: 3.3.2
  @capacitor/core: 3.3.2
```

### Platform(s)

iOS, Android

### Current Behavior

Run `npx cap init`, answer Y to "Migrate Cordova preferences from config.xml?". Any preferences inside a "platform" tag (i.e. `<platform name="android"></platform>` tag, or `<platform name="ios"></platform>`) is ignored.

Furthermore, the documentation does not explain how those platform-specific preferences can be set, whether in capacitor.config.json or elsewhere. The [closest reference](https://capacitorjs.com/docs/plugins/cordova#variables-and-hooks) is

> If your plugin requires variables or settings to be set, you’ll need to apply those configuration settings manually by mapping between the plugin’s plugin.xml and required settings on iOS and Android.

Does this mean manually editing the config.xml in the individual platform directories? That won't work because `npx cap copy` overwrites it.

### Expected Behavior

Cordova supports the [`<platform>` tag](https://cordova.apache.org/docs/en/10.x/config_ref/#platform), which can wrap other elements. There should be a clearly documented way to support platform-specific config in Capacitor that survives `npx cap copy`.

### Additional Context

See #2246

**Comment by milkov85** [comment]
Yep, would be nice to have, now I cannot insert my `APP_SECRET:` depending on platform

**Comment by Lyfei** [comment]
There doesn't seem to be a plan

**Comment by mlynch** [comment]
Take a look at our Trapeze project which can handle many of these tasks: https://trapeze.dev/

**Comment by eric-horodyski** [comment]
This thread has been inactive for some time, and we’re cleaning up older items to keep the backlog focused and manageable.

Details on the effort:
[https://ionic.io/blog/keeping-capacitors-backlog-healthy](https://ionic.io/blog/keeping-capacitors-backlog-healthy)

If this is still relevant today, feel free to open a new issue with current information and we’ll review it.
